pkg/machinery/config/types/v1alpha1/v1alpha1_validation.go

```diff
@@ -117,6 +119,10 @@ func (c *Config) Validate(mode config.RuntimeMode, options ...config.ValidationO
 
 	switch c.Machine().Type() {
 	case machine.TypeInit, machine.TypeControlPlane:
+		if c.Cluster().Etcd().CA() == nil {
```

Comment:
we have `ClusterConfig.Validate()` up there on line 86

Reply:
i didn't want to validate for non control plane machine since there's no etcd there, does that makes sense?

Reply:
we can pass the machine type to the `Validate()` function, but I think it still make sense to validate `EtcdConfig` in a single place in the code